### ISSUE TITLE
MagitPopup without evil bindings

### ISSUE THREAD

**Issue by decoursin** [justbur/evil-magit#8]
As above, MagitPopup (or magit-popup-mode) doesn't have evil bindings yet. However, this works instead so far:
```
(evil-set-initial-state 'magit-popup-mode 'normal)
```
Thanks!

**Comment by justbur** [comment]
The magit popup is mostly TAB and RET, no?

**Comment by decoursin** [comment]
Well, I like everything to have evil mode. For example while within MagitPopup, I can use evil bindings to scroll up and down to see options that I couldn't see otherwise because the screen is too small. Once I'm in it, then, I can use all of evil bindings to easily switch windows, close this window, and all the other stuff my personalized evil  can do.

But the solution above works great, so maybe we don't need to change anything.

**Comment by justbur** [comment]
I wasn't sure there was much value to doing something more with the pop ups
but I'm open to suggestions.

**Comment by decoursin** [comment]
Great thank you. I don't actually have any suggestions.

**Comment by justbur** [comment]
I can do C-bfud for scrolling that shouldn't be a problem. hjkl is a problem because it would mask the magit functions, unless you went to modal editing which I think would be weird in a popup

**Comment by decoursin** [comment]
Yup I'm using modal editing. I have to press i then press the key but I live with it.

**Comment by justbur** [comment]
We could try a leader key approach. Hit the leader key then the option. The magit popups are a little odd in how they implement key bindings though and I haven't completely dug into them yet.

**Comment by decoursin** [comment]
That would be really cool.

**Comment by justbur** [comment]
Try this: 
```elisp
(dolist (char (number-sequence ?! ?z))
  (evil-define-key 'normal magit-popup-mode-map
    (concat ";" (char-to-string char)) 'magit-invoke-popup-action))
```

Be careful though. I'm not positive this will work everywhere, but it allows for `;` as a leader.

**Comment by decoursin** [comment]
Thanks, I gave it a try. It doesn't work for the switches and options, unless I'm doing something wrong. Anyways, ";" as a leader, for me, isn't much better than going into insert mode first.

Ctrl or shift might be better (hold ctrl then click option, for example), but it's probably not worth it. Thank you for giving this attention, but I'm going to close it because it's not worth anymore. I'm satisfied with changing the default initial state and using insert mode.

**Comment by maxsandova** [comment]
I had to set
```
(general-define-key
 :states 'emacs
 :keymaps 'magit-popup-mode-map
 "j" 'forward-button
 "k" 'backward-button
 )
```
In general.el. it would be nice to be able to use emacs-state keybindings for magit-popup-mode but in normal mode, but I guess that would cause a lot of overlapping.
